tests/common.h

```diff
@@ -6,3 +6,38 @@
 // Little helper macro for floating-point comparisons with the units library
 #define BOB_EXPECT_UNIT_T_EQ(val1, val2) \
     EXPECT_DOUBLE_EQ(val1.value(), static_cast<decltype(val1)>(val2).value())
+
+template<class T, class U, class Pred>
+void compareFloatMatricesPredicate(const T &m1, const U &m2, const Pred &compare)
+{
+    static_assert(std::is_same<const float &, const decltype(m1[0]) &>::value &&
+                          std::is_same<const float &, const decltype(m2[0]) &>::value,
+                  "m1 and m2 must contain floats");
+
+    ASSERT_EQ(m1.size(), m2.size());
+
+    SCOPED_TRACE("Comparing matrices");
+    for (int i = 0; i < m1.rows(); i++) {
+        for (int j = 0; j < m1.cols(); j++) {
+            compare(m1(i, j), m2(i, j));
+        }
+    }
+}
+
+template<class T, class U>
+void compareFloatMatrices(const T &m1, const U &m2)
+{
+    const auto expect = [](float a, float b) {
+        EXPECT_FLOAT_EQ(a, b);
```

Comment:
Having been swamped with billions of test errors after writing code like this for GeNN, I'd maybe thing on expecting the result of ``std::all_of`` or something

Reply:
Yeah, I've had similarly verbose error messages :grimacing:. The problem is that we're doing floating-point comparisons so we don't want to test for exact equality.

We could specify the desired precision manually and check that ``a - b < precision`` for all elements, but I'm a bit on the fence about this, because I think Google's default comparison for ``EXPECT_FLOAT_EQ()`` does something clever.

Reply:
Good point and yes, I just looked at the source and it does indeed do _something_ clever 😄